### PR TITLE
Proxy fqdn

### PR DESCRIPTION
**Pull request by rdkdd**

### CODE REVIEW

File: spice/cfg/tests-variants.cfg [hunk 1]
```diff
@@ -165,6 +165,22 @@ variants:
                             ticket_send = 12456
                             include join.cfg
 
+                - 527185:
+                    # only for qemu-kvm-rhev testing
```
Comment (Andrei-Stepanov) [inline spice/cfg/tests-variants.cfg]: why?

Comment (Andrei-Stepanov) [inline spice/cfg/tests-variants.cfg]: why only RHV qemu?

File: spice/cfg/tests-variants.cfg [hunk 2]
```diff
@@ -165,6 +165,22 @@ variants:
...
+                    spice_proxy = http://[2620:52:0:2249:21a:4aff:fe01:46a3]:80
+                    include join.cfg
+
+                - 525961:
+                    spice_proxy = http://10.34.73.1
```
Comment (Andrei-Stepanov) [inline spice/cfg/tests-variants.cfg]: Cartesian config allows to use variables.

We must define this value once at tests.cfg and use  instead specifying IP each time.

Lines to replace:

g 10.34.73.1
./spice-base.cfg:987:###        spice_proxy = http://10.34.73.1:3128
./spice-base.cfg:995:###        spice_proxy = http://10.34.73.1
./tests-variants.cfg:129:                    spice_proxy = http://10.34.73.1:3128
./tests-variants.cfg:151:                    spice_proxy = http://10.34.73.1
./tests-variants.cfg:702:                    spice_proxy = http://10.34.73.1:3128

Define variables

http_proxy_ip = 10.34.73.1
http_proxy_port = 3128

And use them

File: spice/cfg/tests-variants.cfg [hunk 3]
```diff
@@ -165,6 +165,22 @@ variants:
...
+                        role-guest:
+                            listening_addr = "ipv6"
+                            spice_ipv4 = no
+                    include join.cfg
+
+                - 546907:
+                    spice_proxy = http://proxy.spice.brq.redhat.com
```
Comment (Andrei-Stepanov) [inline spice/cfg/tests-variants.cfg]: same here:
http_proxy_fqdn = proxy.spice.brq.redhat.com

File: spice/lib/vm_actions_rv.py [hunk 4]
```diff
@@ -350,23 +350,24 @@ def rv_chk_con(vmi):
         hostname = socket.gethostname()    # See rv_url() function
         remote_ip = socket.gethostbyname(hostname)
     elif cfg.spice_proxy:
-        proxy_port = "3128"
-        if "http" in cfg.spice_proxy:
-            split = cfg.spice_proxy.split('//')[1].split(':')
+        prx_parse = cfg.spice_proxy.split('//')[-1]
+        if ']:' in prx_parse or '.' in prx_parse and ':' in prx_parse:
```
Comment (Andrei-Stepanov) [inline spice/lib/vm_actions_rv.py]: Maybe put here example in comment to easy understand what kind of info your are trying to parse:

# If prx_parse has form: http://[ipv6]:3218

File: spice/lib/vm_actions_rv.py [hunk 5]
```diff
@@ -350,23 +350,24 @@ def rv_chk_con(vmi):
...
+            remote_ip, _, proxy_port = prx_parse.rpartition(':')
         else:
-            split = cfg.spice_proxy.split(':')
-        remote_ip = split[0]
-        if len(split) > 1:
-            proxy_port = split[1]
-        logger.info("Proxy port to inspect: %s", proxy_port)
+            remote_ip, proxy_port = prx_parse, "3128"
+        remote_ip = remote_ip.strip('[]')
+        if remote_ip.split('.')[-1].isalpha():
```
Comment (Andrei-Stepanov) [inline spice/lib/vm_actions_rv.py]: remote_ip == 1.2.3.4
.split('.') == [1,3,4,5]
[-1] == 5
"5".isalpha() ????????? It is not obvious why you check the last after dot is number,

If you need to check IP python has for that functions.

http://stackoverflow.com/questions/319279/how-to-validate-ip-address-in-python


Some very fancy check.  Necessary to re-write.

File: spice/cfg/tests-variants.cfg (hunk 1 above)
Comment (Andrei-Stepanov) [inline spice/cfg/tests-variants.cfg]: Just add that RHEL qemu can be old... or.. smth that

Comment (rdkdd) [inline spice/cfg/tests-variants.cfg]: no qemu-kvm is not listening on ipv6. Only qemu-kvm-rhev.

Comment (Andrei-Stepanov) [inline spice/cfg/tests-variants.cfg]: So, make this knowledge available to all )
# As for Feb 2017  RHEL 7 qemu doesn't listen on IPv6.
Great comment!)

File: spice/lib/vm_actions_rv.py (hunk 5 above)
Comment (rdkdd) [inline spice/lib/vm_actions_rv.py]: I have spent good deal of time on this and verified on few test cases.  https://github.com/spiceqa/tp-spice/pull/56#discussion-diff-101510036R359 is for testing whether addresss in config is fqdn and I found he checking of TLD subscript as the easier check.

Comment (Andrei-Stepanov) [inline spice/lib/vm_actions_rv.py]: Ok, then maybe add some comment?  For me it is not clear why use this approach.

**Comment by rdkdd** [comment]
thanks

**Pull request merged**